lib/foreman_ansible_core/playbook_runner.rb

```diff
@@ -94,5 +97,41 @@ def initialize_ansible_dir(ansible_dir)
         !ansible_dir.nil? && File.exist?(ansible_dir)
       @ansible_dir = ansible_dir
     end
+
+    def found_in_known_hosts?
+      JSON.parse(@inventory)['all']['hosts'].each do |host|
+        return false if Net::SSH::KnownHosts.search_for(host).empty?
+      end
+      true
+    end
+
+    def add_to_known_hosts
```

Comment:
Would it make sense to split this method into a scanner and one that only adds the key to the known hosts files so you can test those functions independently?

Reply:
Yeah, especially since I'll change the scanner to do it through ruby entirely.